johnsonandjohnson/Bodiless-JS
# feat: Fully disable save via env var

## Changes
- Add checks in backend to prevent writing data when save disabled
- Disable image upload and page operations when save disabled
- Remove logic in gatsby-theme-bodiless for setting save-enabled and commit-enabled env vars, and re-implement this logic on both test-site and cxstarter (but logic changed to allow commits on all branches but main, and disallow save on main).

## Test Instructions
- Try on feature branch, behavior should be as now except that push is available on all branches but main (not just changeset)
- Try on main branch (do this locally, check out main branch and reset to this branch) - validate that 
- - image upload is disabled
- page operations are disabled
- content changes do not persist across page reloads
- Using postman or curl, validate that all backend endpoints which write data to filesystem return an error.
- Validate fix for #1479 

## Related Issues
Closes #1522 
Fixes #1479 
-
